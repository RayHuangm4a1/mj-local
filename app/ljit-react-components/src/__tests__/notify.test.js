import React from 'react';
import { shallow, mount, } from 'enzyme';
import Notify, { NorifyContent, generator, } from '../components/notify';

jest.mock('antd/lib/message', () => {
	return {
		open: () => () => {},
	};
});
jest.mock('antd/lib/icon', () => function render() { return <div />; });
jest.mock('antd/lib/button');

jest.useFakeTimers();

describe('Notify', () => {
	it('should success be defined', () => {
		expect(Notify.success).toBeDefined();
	});

	it('should error be defined', () => {
		expect(Notify.error).toBeDefined();
	});

	describe('when duration less than zero', () => {
		it('should throw an error', () => {
			const duration = -500;
			const notify = generator('success');
			const content = 'notify content';

			expect(() => {
				notify(content, duration);
			}).toThrowError(/^Norify: duration must greater than zero.$/);
		});
	});
	describe('when duration equals zero', () => {
		it('should return a function', () => {
			const duration = 0;
			const notify = generator('success');
			const content = 'notify content';

			const result = notify(content, duration);

			expect(result).toBeInstanceOf(Function);
		});
	});
	describe('when duration greater than zero', () => {
		it('should setTmieout have been called', () => {
			const duration = 5000;
			const notify = generator('success');
			const content = 'notify content';

			notify(content, duration);

			expect(setTimeout).toHaveBeenCalledTimes(1);
			expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), duration);
		});
	});

	describe('Notify Type Enums', () => {
		it('should be defined', () => {
			expect(Notify.NotifyTypeEnums).toBeDefined();
		});

		it('should match NorifyContent\'s NotifyTypeEnums', () => {
			expect(Notify.NotifyTypeEnums).toMatchObject(NorifyContent.NotifyTypeEnums);
		});
	});
});

describe('Norify Content', () => {
	it('should handle default props', () => {
		const {
			notifyType,
			onClickClose,
		} = NorifyContent.defaultProps;

		expect(notifyType).toEqual(NorifyContent.NotifyTypeEnums.SUCCESS);
		expect(onClickClose).toBeDefined();
		expect(onClickClose).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const notifyType = NorifyContent.NotifyTypeEnums.ERROR;
		const content = 'notify content';
		const onClickClose = () => {};
		const wrapper = shallow(
			<NorifyContent
				notifyType={notifyType}
				content={content}
				onClickClose={onClickClose}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-notify', () => {
		const wrapper = shallow(<NorifyContent />);

		expect(wrapper.hasClass('ljit-notify')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const notifyType = NorifyContent.NotifyTypeEnums.ERROR;
		const content = 'notify content';
		const onClickClose = () => {};
		const wrapper = mount(
			<NorifyContent
				notifyType={notifyType}
				content={content}
				onClickClose={onClickClose}
			/>
		);

		expect(wrapper.props().notifyType).toEqual(notifyType);
		expect(wrapper.props().content).toEqual(content);
		expect(wrapper.props().onClickClose).toEqual(onClickClose);
	});

	describe('Notify Type Enums', () => {
		it('should contains success property', () => {
			const type = 'success';

			expect(NorifyContent.NotifyTypeEnums.SUCCESS).toEqual(type);
		});

		it('should contains error property', () => {
			const type = 'error';

			expect(NorifyContent.NotifyTypeEnums.ERROR).toEqual(type);
		});

		it('should contains info property', () => {
			const type = 'info';

			expect(NorifyContent.NotifyTypeEnums.INFO).toEqual(type);
		});
	});
});
