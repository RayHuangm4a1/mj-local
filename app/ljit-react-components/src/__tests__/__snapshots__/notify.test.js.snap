// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Norify Content should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotifyContent
    content="notify content"
    notifyType="error"
    onClickClose={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          className="ljit-notify__icon"
          color="danger"
          isTranslucent={false}
          size="middle"
          spin={false}
          style={Object {}}
          theme="filled"
          type="close-circle"
        />,
        <div
          className="ljit-notify__content"
        >
          notify content
        </div>,
        <IconButton
          className="ljit-notify__close"
          disabled={false}
          onClick={[Function]}
          type="close"
        />,
      ],
      "className": "ljit-notify ljit-notify--error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ljit-notify__icon",
          "color": "danger",
          "isTranslucent": false,
          "size": "middle",
          "spin": false,
          "style": Object {},
          "theme": "filled",
          "type": "close-circle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "notify content",
          "className": "ljit-notify__content",
        },
        "ref": null,
        "rendered": "notify content",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ljit-notify__close",
          "disabled": false,
          "onClick": [Function],
          "type": "close",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            className="ljit-notify__icon"
            color="danger"
            isTranslucent={false}
            size="middle"
            spin={false}
            style={Object {}}
            theme="filled"
            type="close-circle"
          />,
          <div
            className="ljit-notify__content"
          >
            notify content
          </div>,
          <IconButton
            className="ljit-notify__close"
            disabled={false}
            onClick={[Function]}
            type="close"
          />,
        ],
        "className": "ljit-notify ljit-notify--error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ljit-notify__icon",
            "color": "danger",
            "isTranslucent": false,
            "size": "middle",
            "spin": false,
            "style": Object {},
            "theme": "filled",
            "type": "close-circle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "notify content",
            "className": "ljit-notify__content",
          },
          "ref": null,
          "rendered": "notify content",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ljit-notify__close",
            "disabled": false,
            "onClick": [Function],
            "type": "close",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
